spec/system/s3_uploads_spec.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+describe "Uploading files to S3", type: :system do
+  fab!(:current_user) { Fabricate(:admin) }
+
+  let(:modal) { PageObjects::Modals::Base.new }
+  let(:composer) { PageObjects::Components::Composer.new }
+
+  describe "direct S3 uploads" do
+    before { SiteSetting.enable_direct_s3_uploads = true }
+
+    describe "single part uploads" do
+      it "uploads custom avatars to S3" do
+        skip_s3_system_spec?
+
+        setup_s3_system_test
+        sign_in(current_user)
+
+        visit "/my/preferences/account"
+
+        find("#edit-avatar").click
+        find("#uploaded-avatar").click
+        attach_file(File.absolute_path(file_from_fixtures("logo.jpg"))) do
+          find("#avatar-uploader").click
+        end
+        expect(page).to have_css(".avatar-uploader label[data-uploaded]")
+        modal.click_primary_button
+        expect(page).to have_css(
+          "#user-avatar-uploads[data-custom-avatar-upload-id]",
+          visible: false,
+        )
+        puts page.driver.browser.logs.get(:browser).map(&:message)
+        expect(current_user.reload.uploaded_avatar_id).to eq(
+          find("#user-avatar-uploads", visible: false)["data-custom-avatar-upload-id"].to_i,
+        )
+      end
+    end
+
+    describe "multipart uploads" do
+      it "uploads a file in the post composer" do
+        skip_s3_system_spec?
+
+        setup_s3_system_test
+        sign_in(current_user)
+
+        visit "/new-topic"
+
+        file_path = file_from_fixtures("logo.png", "images").path
+        attach_file(file_path) { composer.click_toolbar_button("upload") }
+
+        expect(page).to have_no_css("#file-uploading")
```

Comment:
When running locally, it takes aaaaaages for the upload to finish, so this check times-out.

This is what I see in the JS console:

<img width="538" alt="SCR-20230815-twqx" src="https://github.com/discourse/discourse/assets/6270921/5c68a7ad-1107-4e3f-9ca0-f3f91cb80b7a">

Kinda suspicious that all the durations are almost exactly multiples of 5 seconds 🤔 

Anything else I can do to help diagnose what's making it take so long?

Reply:
Ah never mind, I see the macOS issue noted at https://github.com/discourse/minio_runner#hosts-file 👌

Reply:
@davidtaylorhq yeah that macOS issue was driving me nuts, such a weird thing to happen. Glad you found it :)